### ISSUE TITLE
versioning is overly tight

### ISSUE THREAD

**Issue by jonassmedegaard** [joepio/atomic-data-rust#329]
The `Cargo.toml` files declare dependencies with single versions.

That's overly tight: `Cargo.toml` files should declare the widest possible range of supported versions, whereas `Cargo.lock` tracks the specific combination of versions tested well together.

Crates generally (if not always?) follow [Semantic Versioning](https://semver.org/) which essentially means that minor version is bumped at each API extension and major version is bumped at each API breakage.  For consuming projects it means that if you use version 1.2.3 then you can rest assured that version 1.999.999 will work as well, and you can therefore declare that range like this: `^1.2.3`.

In short: Please append `^` to all version numbers in dependencies in `Cargo.toml` files, except where you know for certain that you can only use one very specific single version.

**Comment by joepio** [comment]
In Rust, the `^` symbol is optional - every version is _implicitly_ ranged, so adding this would not have any impact.

However, when I specify `1.2.3` instead of `1.2`, I may be overly specific / tight. I'll be sure to keep this in mind, and I'll try a build with a bit less specific versions soon.

**Comment by jonassmedegaard** [comment]
> In Rust, the ^ symbol is optional

Oh, I was unaware of that. Thanks for that detail!

**Comment by jonassmedegaard** [comment]
> However, when I specify 1.2.3 instead of 1.2, I may be overly specific / tight.

Yeah, but I would consider that a less important issue: Fine if you notice that further relaxing is possible, but if you're busy then I suggest you simply wait for consumers of your project to eventually complain (and have them verify compliance) and spend your attention elsewhere :-)
